### PR TITLE
Dil sigs

### PR DESCRIPTION
**Pull request by algonathan**

### CODE REVIEW

File: crypto/dillithium_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		dsigner := NewDilithiumSigner()
+		b := make([]byte, 8)
+		binary.BigEndian.PutUint64(b, uint64(i))
+		bs := sha256.Sum256(b)
+		sig := dsigner.SignBytes(bs[:])
+		//sig := dil2Sign(sk, bs[:])
+		dvf := dsigner.GetVerifyingKey()
+		dverifier, err := dvf.GetVerifier()
+		a.NoError(err)
+		a.NoError(dverifier.VerifyBytes(bs[:], sig))
```
Comment (algoidan) [inline crypto/dillithium_test.go:39]: we can also add verifybytes and sign on hashable tests

File: crypto/dillithium.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	return d.SignBytes(hs[:])
+}
+
+// SignBytes receives bytes and signs over them.
+// the size of the signature should conform with dil2Signature.
+func (d *DilithiumSigner) SignBytes(data []byte) ByteSignature {
+	return d.Keypair.SignBytes(data)
+}
+
+// GetVerifyingKey Outputs a veryfying key ovject which is serializeable.
```
Comment (algoidan) [inline crypto/dillithium.go:61]: typo - object

**Review by algoidan — COMMENTED** [review]
looks great overall!
we should also use Dilithium cat. 3

File: crypto/internal/cdilithium/wrapper.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	return sig[:]
+}
+
+// ErrBadDilithiumSignature indicates signature isn't valid.
+var ErrBadDilithiumSignature = errors.New("bad signature")
+
+// VerifyBytes follows dilithium algorithm to verify a signature.
+func (v *Dil2PublicKey) VerifyBytes(data []byte, sig []byte) error {
+	cdata := (*C.uchar)(C.NULL)
+	if len(data) != 0 {
```
[the remark this answers is not in the document]
Comment (algonathan) [reply crypto/internal/cdilithium/wrapper.go]: Good catch! I've added a test that checks an empty signature. the code panicked - I've added a check in the code too.

File: crypto/dillithium_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	a := require.New(t)
+	dsigner := NewDilithiumSigner()
+
+	sig := dsigner.Sign(TestingHashable{})
+
+	dvf := dsigner.GetVerifyingKey()
+	dverifier, err := dvf.GetVerifier()
+	a.NoError(err)
+	a.NoError(dverifier.Verify(TestingHashable{}, sig))
+	a.Error(dverifier.Verify(TestingHashable{
```
Comment (algoidan) [inline crypto/dillithium_test.go:53]: can you please add one test that verifies non empty hashable

File: crypto/dillithium_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+	bs := sha256.Sum256(make([]byte, 8))
+	sig2 := dsigner.SignBytes(bs[:])
+	a.NoError(dverifier.VerifyBytes(bs[:], sig2))
+
+	bs2 := bs
+	bs2[0]++
+	a.Error(dverifier.VerifyBytes(bs2[:], sig2))
+	sig2[0]++
+	a.Error(dverifier.VerifyBytes(bs[:], sig2))
```
Comment (algoidan) [inline crypto/dillithium_test.go:67]: the bs is also wrong here. you should copy the slice

**Review by algoidan — COMMENTED** [review]
two small things. looks great overall

**Pull request merged**